moto/awslambda/models.py

```diff
@@ -667,6 +669,28 @@ def put_function(self, fn):
         :param fn: Function
         :type fn: LambdaFunction
         """
+        pattern = r"arn:(aws[a-zA-Z-]*)?:iam::(\d{12}):role/?[a-zA-Z_0-9+=,.@\-_/]+"
+        valid_role = re.match(pattern, fn.role)
+        if valid_role:
+            account = valid_role.group(2)
+            if account != ACCOUNT_ID:
+                raise ValueError(
```

Comment:
Please add the specific exceptions to the corresponding `exceptions.py` file to implement one of the moto exception classes.